snuba/query/validation/signature.py

```diff
@@ -112,9 +111,11 @@ def validate(self, expression: Expression, schema: ColumnSet) -> None:
             return
 
         column_type = column.type.get_raw()
-        nullable = NullableOld in column.type.get_all_modifiers() or isinstance(
-            column.type, Nullable
-        )
+
+        nullable = isinstance(column_type, Nullable)
+
+        if nullable:
+            column_type = cast(Nullable, column_type).inner_type
```

Comment:
It seems this will be a common pattern. Should we move it into the Nullable class itself or should we wait for when Nulalble will be an attribute of a column type ?

Reply:
Which part do you mean to move into the Nullable class? The `cast` is annoying but seems kind of unavoidable without changing the whole ColumnType. I don't think it will be a common pattern, this is the only place where we need to do this.

Reply:
Actually nevermind, after the clarification above, I think it is fair that whoever deals with types will have to deal with Nullable because it is always a structural property of the type and not a modifier the query processing code can ignore (like materialized).